# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/avr32/package/.../x-arch-arvr32-sidsafb.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./drivers/video/sidsafb.c.vanilla	2007-03-08 17:27:08.436961748 +0000
+++ ./drivers/video/sidsafb.c	2007-03-09 11:15:05.354758556 +0000
@@ -274,15 +274,18 @@
 	/* Force same alignment for each line */
 	var->xres = (var->xres + 3) & ~3UL;
 	var->xres_virtual = (var->xres_virtual + 3) & ~3UL;
+	var->xoffset = var->yoffset = 0;
+
+	var->red.msb_right = var->green.msb_right = var->blue.msb_right =
+		var->transp.msb_right = 0;
 
-	var->red.msb_right = var->green.msb_right = var->blue.msb_right = 0;
 	var->transp.offset = var->transp.length = 0;
 
 	switch (var->bits_per_pixel) {
 	case 2:
 	case 4:
 	case 8:
-		var->red.offset = var->green.offset = var->blue.offset = 0;
+		var->green.offset = var->blue.offset = 0;
 		var->red.length = var->green.length = var->blue.length
 			= var->bits_per_pixel;
 		break;
@@ -292,7 +295,6 @@
 		 * Bit 16 is the "intensity" bit, I think. Not sure
 		 * what we're going to use that for...
 		 */
-		var->red.offset = 0;
 		var->green.offset = 5;
 		var->blue.offset = 10;
 		var->red.length = 5;
@@ -304,9 +306,8 @@
 		var->transp.length = 8;
 		/* fall through */
 	case 24:
-		var->red.offset = 16;
 		var->green.offset = 8;
-		var->blue.offset = 0;
+		var->blue.offset = 16;
 		var->red.length = var->green.length = var->blue.length = 8;
 		break;
 	default:
@@ -315,10 +316,6 @@
 		return -EINVAL;
 	}
 
-	var->xoffset = var->yoffset = 0;
-	var->red.msb_right = var->green.msb_right = var->blue.msb_right =
-		var->transp.msb_right = 0;
-
 	return 0;
 }
 
